juju/juju
# fix: remove update trigger with only primary keys

Some tables have nothing but primary keys. We don't support updating the primary keys with the change stream, so that is undefined behaviour. If there is a table with only primary keys, there isn't a way to update the table, so we can then drop the update trigger.

In addition, ensuring that the change_log_namspace id and the trigger namespace code int match is fraught with danger. It's too easy to get it wrong and then the changelog stream breaks. The solution is to insert the changestream namespace at the same register time as the trigger. This ensures that they're both the same and provides better guard rails.

## Checklist





## QA steps

```
$ juju bootstrap lxd test
$ juju add-model default
$ juju deploy ubuntu
```

All of the domain tests should pass.



## Links


**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/

**Jira card:** JUJU-
